src/yunohost/app.py

```diff
@@ -1601,6 +1604,23 @@ def app_config_apply(operation_logger, app, args):
                 generated_name = ("YNH_CONFIG_%s_%s_%s" % (tab_id, section_id, option_name)).upper()
 
                 if generated_name in args:
+                    # Upload files from API
+                    # A file arg contains a string with "FILENAME:BASE64_CONTENT"
+                    if option["type"] == "file" and msettings.get('interface') == 'api':
+                        if upload_dir is None:
+                            upload_dir = tempfile.mkdtemp(prefix='tmp_configpanel_')
+                        filename, args[generated_name] = args[generated_name].split(':')
+                        logger.debug("Save uploaded file %s from API into %s", filename, upload_dir)
+                        file_path = os.join(upload_dir, filename)
+                        try:
+                            with open(file_path, 'wb') as f:
```

Comment:
Note that as `filename` is chosen by the user, this gives arbitrary file write (i.e. `os.path.join("/foo", "/etc/passwd") == "/etc/passwd"`) through this API.

Reply:
fixed

Reply:
Does it actually fix the issue?

~~~
>>> os.path.normpath("/tmp/" + "../../etc/passwd")
'/etc/passwd'
~~~